Youtube-Clone/Youtube-Clone/Sources/ViewControllers/SignupVC.swift

```diff
@@ -0,0 +1,169 @@
+        }
+        loginStackView.snp.makeConstraints {
+            $0.top.equalTo(titleLabel.snp.bottom).offset(110)
+            $0.leading.trailing.equalToSuperview().inset(30)
+        }
+        showPasswordButton.snp.makeConstraints {
+            $0.top.equalTo(loginStackView.snp.bottom).offset(10)
+            $0.leading.equalToSuperview().inset(20)
+        }
+        signinButton.snp.makeConstraints {
+            $0.top.equalTo(loginStackView.snp.bottom).offset(70)
+            $0.leading.trailing.equalToSuperview().inset(30)
+            $0.height.equalTo(40)
+        }
+    }
+    
+    private func setupTextfield() {
+        nameTextfield.delegate = self
+        accountTextfield.delegate = self
+        passwordTextfield.delegate = self
+    }
+    
+    private func setupButtonState(_ button: UIButton) {
+        if #available(iOS 15.0, *) {
+            var configuration = UIButton.Configuration.plain()
+            configuration.image = UIImage(systemName: canShow ? "checkmark.square.fill" :  "square")
+            configuration.titlePadding = 10
+            configuration.imagePadding = 10
+            configuration.baseForegroundColor = canShow ? .googleBlue : .lightGray
+            configuration.attributedTitle = AttributedString("비밀번호 표시", attributes: AttributeContainer([NSAttributedString.Key.foregroundColor: UIColor.black, NSAttributedString.Key.font: UIFont.systemFont(ofSize: 13)]))
+            button.configuration = configuration
+        } else {
+            button.setTitle("비밀번호 표시", for: .normal)
+            button.titleLabel?.font = UIFont.systemFont(ofSize: 13)
+            button.setTitleColor(.black, for: .normal)
+            button.setImage(UIImage(systemName: canShow ? "checkmark.square.fill" :  "square"), for: .normal)
+            button.tintColor = canShow ? .googleBlue : .lightGray
+            button.titleEdgeInsets = UIEdgeInsets(top: 0, left: 10, bottom: 0, right: 0)
+        }
+        
+        passwordTextfield.isSecureTextEntry = canShow ? false : true
+    }
+    
+    // MARK: - @objc
+    @objc
+    private func touchUpSignIn() {
+        let vc = CheckVC()
+        if let text = nameTextfield.text {
+            vc.titleLabel.text = text + "님\n환영합니다!"
+        }
+        present(vc, animated: true, completion: nil)
+    }
+    
+    @objc
+    private func touchUpShowPassword() {
+        canShow.toggle()
+        setupButtonState(showPasswordButton)
+    }
+}
+
+// MARK: - UITextFieldDelegate
+extension SignupVC: UITextFieldDelegate {
+    func textFieldDidEndEditing(_ textField: UITextField) {
+        if let isNameEmpty = nameTextfield.text?.isEmpty,
+           let isAccountEmpty = accountTextfield.text?.isEmpty,
+           let isPasswordEmpty = passwordTextfield.text?.isEmpty {
+            if isNameEmpty || isAccountEmpty || isPasswordEmpty {
+                signinButton.isEnabled = false
+                signinButton.backgroundColor = .lightGray
```

Comment:
```suggestion
        if nameTextfield.hasText &&
          accountTextfield.hasText &&
           passwordTextfield.hasText {
              signinButton.isEnabled = false
              signinButton.backgroundColor = .lightGray
```
UITextfield.text?.isEmpty 보다는 UITextfield.hasText 속성을 사용하는 건 어떨까요???

Reply:
요런 지적 너무 좋습니다👍  반영해서 머지할게요!